src/test/java/com/puppycrawl/tools/checkstyle/DetailNodeTreeStringPrinterTest.java

```diff
@@ -37,10 +37,17 @@
 import com.puppycrawl.tools.checkstyle.api.LocalizedMessage;
 import com.puppycrawl.tools.checkstyle.api.TokenTypes;
 
-public class DetailNodeTreeStringPrinterTest {
+public class DetailNodeTreeStringPrinterTest extends AbstractTreeTestSupport {
 
-    private static String getPath(String filename) {
-        return "src/test/resources/com/puppycrawl/tools/checkstyle/astprinter/" + filename;
+    // [REFLECTION]
+    // DetailNodeTreeStringPrinter#getParseErrorMessage is used for creating error messages
+    // for validating those obtained in UTs against the ones created.
+    private static final Method GET_PARSE_ERROR_MESSAGE = Whitebox.getMethod(
+            DetailNodeTreeStringPrinter.class, "getParseErrorMessage", ParseErrorMessage.class);
```

Comment:
@romani , does this look like a utility method? https://github.com/checkstyle/checkstyle/blob/46fa8090954480eecdeb6e7367f0e30655853f42/src/main/java/com/puppycrawl/tools/checkstyle/DetailNodeTreeStringPrinter.java#L91-L101
I think it does. If you agree we should make it public and move to JavadocUtils.
Otherwise reflection usage is reasonable.